src/coreComponents/constitutive/fluid/CompositionalMultiphaseFluid.cpp

```diff
@@ -89,11 +72,21 @@ template< typename ARRAY >
 void checkInputSize( ARRAY const & array, integer const expected, string const & attr, string const & name )
 {
   GEOSX_THROW_IF_NE_MSG( array.size(), expected,
-                         name << ": invalid number of entries in " << attr << " attribute",
+                         GEOSX_FMT( "{}: invalid number of entries in attribute '{}'", name, attr ),
```

Comment:
Here for example, I'm not convinced that `fmt` brings a lot _w.r.t._ `<<`.

I've been using this kind of approach on python for a while and I also think that's it's often more convenient to write strings with the `fmt` approach.
But when it comes to reading, I'm less convinced because your eyes have to swing here and there...

It's not as great as python's `fstring`!

Reply:
I think it still brings value, in the form of decoupling message structure from arguments, allowing you to change the message itself more easily. In this PR I specifically didn't change any of the messages in order to show 1-to-1 old/new comparison, but one can imagine wanting to rewrite messages at some point, and having format strings just makes it easier.

For example, one might want to standardize messages that check for number of arguments in an attribute across the code; you can define a `constexpr` format string in one header and reuse it everywhere. Not to mention that it's much more `grep`-able if you're searching for the source of a user-reported error in code. I do agree that some of the bigger advantages of `[std|fmt]::format` - e.g. type safety/awareness, flexible formats specifiers - don't shine in this particular example where we're just dealing with strings.

As for f-strings, I love them in Python, but it's unlikely to ever happen in C++, sadly.

Reply:
> ...one might want to standardize messages that check for ...

This is a relevant argument.
I would nevertheless say that "_same error message_" -> "_same error_" -> "_single function_".